Firestore/core/src/firebase/firestore/core/query.cc

```diff
@@ -41,9 +66,11 @@ bool Query::MatchesPath(const Document& doc) const {
   }
 }
 
-bool Query::MatchesFilters(const Document&) const {
-  // TODO(rsgowman): Implement this correctly.
-  return true;
+bool Query::MatchesFilters(const Document& doc) const {
+  return std::all_of(filters_.begin(), filters_.end(),
```

Comment:
Agree it's not worth the effort for just this one usage in this PR. I'm hoping we'll import this at some point, though, because it will come up almost every time a standard algorithm is used.